src/Transport/Topology/Topologies/ForwardingTopologySectionManager.cs

```diff
@@ -89,30 +98,32 @@ public TopologySectionInternal DetermineResourcesToCreate(QueueBindings queueBin
             };
         }
 
-        public TopologySectionInternal DeterminePublishDestination(Type eventType)
+        public async Task<TopologySectionInternal> DeterminePublishDestination(Type eventType)
```

Comment:
@danielmarbach I'm not convinced it's needed.
If you still think that we should, let discuss discuss on a call.